src/Trace/Propagator/B3HeadersPropagator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace OpenCensus\Trace\Propagator;
+
+use OpenCensus\Trace\SpanContext;
+
+/**
+ * @see https://github.com/openzipkin/b3-propagation
+ */
+class B3HeadersPropagator implements PropagatorInterface
+{
+    private const X_B3_TRACE_ID = 'X-B3-TraceId';
+    private const X_B3_SPAN_ID = 'X-B3-SpanId';
+    private const X_B3_SAMPLED = 'X-B3-Sampled';
+    private const X_B3_FLAGS = 'X-B3-Flags';
+
+    /**
+     * Extract the SpanContext from some container
+     *
+     * @param mixed $container
+     * @return SpanContext
+     */
+    public function extract($container)
+    {
+        $traceId = $container[self::X_B3_TRACE_ID] ?? null;
+        $spanId = $container[self::X_B3_SPAN_ID] ?? null;
+        $sampled = $container[self::X_B3_SAMPLED] ?? null;
+        $flags = $container[self::X_B3_FLAGS] ?? null;
+
+        $enabled = null;
+
+        if ($sampled !== null) {
+            $enabled = $sampled === '1';
```

Comment:
It is possible some propagators pass `true` aswell according to https://github.com/openzipkin/b3-propagation#sampling-state-1